$MirOS: ports/databases/postgresql/patches/patch-src_backend_access_transam_xact_c,v 1.2 2007/02/22 20:51:01 tg Exp $
--- src/backend/access/transam/xact.c.orig	Tue Mar  4 19:54:00 2008
+++ src/backend/access/transam/xact.c	Wed Aug 27 10:23:01 2008
@@ -4256,7 +4256,7 @@ xact_desc_commit(StringInfo buf, xl_xact
 	int			i;
 
 	appendStringInfo(buf, "%04u-%02u-%02u %02u:%02u:%02u",
-					 tm->tm_year + 1900, tm->tm_mon + 1, tm->tm_mday,
+					 (unsigned)(tm->tm_year + 1900), tm->tm_mon + 1, tm->tm_mday,
 					 tm->tm_hour, tm->tm_min, tm->tm_sec);
 	if (xlrec->nrels > 0)
 	{
@@ -4287,7 +4287,7 @@ xact_desc_abort(StringInfo buf, xl_xact_
 	int			i;
 
 	appendStringInfo(buf, "%04u-%02u-%02u %02u:%02u:%02u",
-					 tm->tm_year + 1900, tm->tm_mon + 1, tm->tm_mday,
+					 (unsigned)(tm->tm_year + 1900), tm->tm_mon + 1, tm->tm_mday,
 					 tm->tm_hour, tm->tm_min, tm->tm_sec);
 	if (xlrec->nrels > 0)
 	{
